### PR TITLE
#331 Fix retrieving golang dependencies

### LINKED ISSUE #331: Fix Go dependencies
Currently project-keeper's java class `GolangServices` uses
```
go list -f "{{if not .Indirect}}{{.}}{{end}}" -m all
``` 

and applies `go-licenses csv` to each entry.

This fails for repository [project-overview](https://github.com/eaxasol/project-overview), which currently has dependencies to 
* [github.com/stretchr/testify/assert](https://github.com/stretchr/testify/assert) and 
* [github.com/google/go-github/v45/github](https://github.com/google/go-github/v45/github)

To fix project-keeper go list needs to be called like this:
```
go list -f "{{if .Module}}{{if not .Module.Indirect}}{{.ImportPath}} {{.Module.Version}}{{end}}{{end}}" all
```

See also https://pkg.go.dev/cmd/go#hdr-List_packages_or_modules

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #331

### CODE REVIEW

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java [hunk 1]
```diff
@@ -57,17 +59,37 @@ private List<DependencyChange> getDependencyChanges(final Type type) {
...
-                        name, this.dependencies.getDependencies()).ticketMitigation().toString()));
+                .orElseGet(() -> getTypeByPrefix(moduleName));
+    }
+
+    private Type getTypeByPrefix(final String moduleName) {
+        if (moduleName.matches(".*/v\\d+")) {
+            final String moduleNamePrefix = moduleName.substring(0, moduleName.lastIndexOf("/"));
+            LOGGER.finest(() -> "Found prefix '" + moduleNamePrefix + "' for module '" + moduleName + "'.");
+            final Optional<Type> type = this.dependencies.getDependencies().stream() //
+                    .filter(dep -> dep.getName().startsWith(moduleNamePrefix)) //
```
Comment (tomuben) [inline project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java:78]: is `startsWith()`  really safe?

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangServices.java [hunk 2]
```diff
@@ -82,6 +81,20 @@ Map<String, GolangDependencyLicense> getLicenses(final Path absoluteSourcePath,
...
 
+    Path getModuleDir(final Path absoluteSourcePath, final String moduleName) {
+        final SimpleProcess process = GoProcess.start(absoluteSourcePath,
+                List.of("go", "list", "-m", "-f", "{{.Dir}}", moduleName));
+        process.waitUntilFinished(Duration.ofSeconds(3));
+        final Path path = Paths.get(process.getOutputStreamContent().trim());
+        if (!Files.exists(path)) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-PK-CORE-156")
+                    .message("Error directory {{directory}} for module {{module name}} does not exist", path,
+                            moduleName)
```
Comment (tomuben) [inline project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangServices.java:92]:
```suggestion
                    .message("Directory {{directory}} for module {{module name}} does not exist", path,
                            moduleName)
```

**Review by tomuben — COMMENTED** [review]
please check my comments

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java [hunk 3]
```diff
@@ -57,17 +59,37 @@ private List<DependencyChange> getDependencyChanges(final Type type) {
...
+                    .map(ProjectDependency::getType) //
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java:79]: They force a line break in the automatic formatter. Looks strange but easier to read without :-P

File: project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java (hunk 1 above)
Comment (kaklakariada) [reply project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java:78]: What do you mean with safe? It's a simple solution that works.

**Review by ckunki — APPROVED** [review]

**Pull request merged**